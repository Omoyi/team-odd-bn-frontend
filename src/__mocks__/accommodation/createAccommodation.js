// create accommodation reducers
import {
  NEW_ACCOMMODATION_INPUT,
  NEW_ADDED_PHOTOS,
  OPEN_ROOMS_MODAL,
  NEW_ADDED_ROOMS,
  GET_CITIES_SUCCESS,
  CREATE_ACCOMMODATION_SUCCESS,
  CREATE_ACCOMMODATION_ERROR,
  SPINNER_STATUS,
} from '../../redux/actionTypes/accommodationActionTypes';

export const accommodationInputAction = {
  type: NEW_ACCOMMODATION_INPUT,
  payload: {
    name: 'Kilimanjaro',
    city: 3,
    description: 'Nice place to rest in the world',
    address: 'KG 208 st',
    imageUrls: ['me.jpg', 'you.png'],
    rooms: [
      {
        name: 'Kagera',
        roomType: 'Single',
      },
      {
        name: 'Kakum',
        roomType: 'Double',
      },
    ],
  },
};
export const photosInputAction = {
  type: NEW_ADDED_PHOTOS,
  photos: 'me.you',
};
export const openModalAction = {
  type: OPEN_ROOMS_MODAL,
  openRoomsModal: true,
};
export const addNewRoomsAction = {
  type: NEW_ADDED_ROOMS,
  rooms: 'Akagera',
};
export const getCitiesSuccessAction = {
  type: GET_CITIES_SUCCESS,
  retrievedCities: {
    message: 'Location retrieved',
    data: [{
      id: 1,
      city: 'Nairobi',
      createdAt: '2019-12-09T07:36:36.417Z',
      updatedAt: '2019-12-09T07:36:36.417Z',
    }],
  },
};

export const createAccommodation = {
  type: CREATE_ACCOMMODATION_SUCCESS,
  payload: {
    name: 'Mubarak Palace',
    cityId: 3,
    address: 'fghj',
    googleCoordinates: '-77, -56',
    imageUrls: ['l.jpg', 'k.jpg'],
    description:
             ' Queens palace is the best accomodation facility you could ever get',
    rooms: [
      { name: '12', roomType: 'single' },
      { name: '13', roomType: 'single' },
      { name: '14', roomType: 'single' },
    ],
  },
};
export const createAccommodationError = {
  type: CREATE_ACCOMMODATION_ERROR,
  status: 'error',
  error: 'You are not a travel administrator',
};

export const spinnerStatus = {
  type: SPINNER_STATUS,
  spinner: false,
};

// create accommodation action

export const retrievedCitiesAction = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: GET_CITIES_SUCCESS,
    retrievedCities: {
      message: 'Location retrieved',
      data: [
        {
          id: 1,
          city: 'Kampala',
          createdAt: '2019-12-09T07:36:36.417Z',
          updatedAt: '2019-12-09T07:36:36.417Z',
        },
      ],
    },
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'success',
    error: {},
  },
];
export const retrievedCitiesApi = {
  status: 200,
  response: {
    message: 'Location retrieved',
    data: [
      {
        id: 1,
        city: 'Kampala',
        createdAt: '2019-12-09T07:36:36.417Z',
        updatedAt: '2019-12-09T07:36:36.417Z',
      },
    ],
  },
};

export const tokenErrorApiRequest = {
  status: 400,
  response: {
    message: 'JWT malformed',
  },
};

export const createAccommodationApi = {
  status: 201,
  response: {
    message: 'Accommodation created',
    data: {
      id: 24,
      userId: 15,
      name: "Jox's palace lo",
      cityId: 3,
      address: 'fghj',
      description:
               ' Queens palace is the best accomodation facility you could ever get',
      googleCoordinates: '-77, -56',
      updatedAt: '2020-01-15T18:17:26.264Z',
      createdAt: '2020-01-15T18:17:26.264Z',
      imageUrls: [
        {
          id: 34,
          accommodationId: 24,
          imageUrl: 'l.jpg',
          updatedAt: '2020-01-15T18:17:26.277Z',
          createdAt: '2020-01-15T18:17:26.277Z',
        },
      ],
      rooms: [
        {
          id: 46,
          name: 'Mugwizabatware',
          roomType: 'single',
          accommodationId: 24,
          updatedAt: '2020-01-15T18:17:26.318Z',
          createdAt: '2020-01-15T18:17:26.318Z',
        },
      ],
    },
  },
};
export const createAccommodationMockAction = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_SUCCESS,
    payload: {
      message: 'Accommodation created',
      data: {
        id: 24,
        userId: 15,
        name: "Jox's palace lo",
        cityId: 3,
        address: 'fghj',
        description:
                 ' Queens palace is the best accomodation facility you could ever get',
        googleCoordinates: '-77, -56',
        updatedAt: '2020-01-15T18:17:26.264Z',
        createdAt: '2020-01-15T18:17:26.264Z',
        imageUrls: [
          {
            id: 34,
            accommodationId: 24,
            imageUrl: 'l.jpg',
            updatedAt: '2020-01-15T18:17:26.277Z',
            createdAt: '2020-01-15T18:17:26.277Z',
          },
        ],
        rooms: [
          {
            id: 46,
            name: 'Mugwizabatware',
            roomType: 'single',
            accommodationId: 24,
            updatedAt: '2020-01-15T18:17:26.318Z',
            createdAt: '2020-01-15T18:17:26.318Z',
          },
        ],
      },
    },
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'success',
    error: {},
  },
];
export const createAccommodationMockError = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'error',
    error: {
      message: 'You are not a travel admnistrator',
    },
  },
];
export const createAccommodationMockApiError = {

  status: 401,
  response: {
    data: {
      message: 'You are not a travel admnistrator',
    },
  },
};
export const createAccommodationMockApiErrors = {
  status: 401,
  response: {
    data: {
      message: [
        'You are not a travel admnistrator',
        'imageURL should be valid',
      ],
    },
  },
};
export const createAccommodationMockErrors = [
  {
    spinner: false,
    type: SPINNER_STATUS,
  },
  {
    type: CREATE_ACCOMMODATION_ERROR,
    status: 'error',
    error: {
      message: [
        'You are not a travel admnistrator',
        'imageURL should be valid',
      ],
    },
  },
];
// Props
export const createAccommodationProps = {
  stateObject: {
    accommodation: {
      createAccommodation: {
        spinner: false,
        openRoomsModal: false,
        retrievedCities: {
          data: [
            {
              id: 1,
              city: 'Kampala',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
            {
              id: 2,
              city: 'Kigali',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
          ],
        },
        payload: {
          message: 'Accommodation posted successfully',
        },
      },
    },
  },
  createAccommodationAction: jest.fn(),
  newAccommodationInput: jest.fn(),
  newAccommodationPhotos: jest.fn(),
  roomsModalAction: jest.fn(),
  getCitiesAction: jest.fn(),
  updateSpinnerStatus: jest.fn(),
};
export const modalSpinnerProps = {
  stateObject: {
    accommodation: {
      createAccommodation: {
        spinner: true,
        openRoomsModal: true,
        retrievedCities: {
          data: [
            {
              id: 1,
              city: 'Kampala',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
            {
              id: 2,
              city: 'Kigali',
              createdAt: '2019-12-09T07:36:36.417Z',
              updatedAt: '2019-12-09T07:36:36.417Z',
            },
          ],
        },
      },
    },
  },
  createAccommodationAction: jest.fn(),
  newAccommodationInput: jest.fn(),
  newAccommodationPhotos: jest.fn(),
  roomsModalAction: jest.fn(),
  getCitiesAction: jest.fn(),
  updateSpinnerStatus: jest.fn(),
};
export const mainState = {
  accommodation: {
    createAccommodation: {
      rooms: [],
      photos: [],
    },
  },
};

export const modalProps = {
  appendRow: jest.fn(),
  addNewRoomsAction: jest.fn(),
  roomsModalAction: jest.fn(),
};
