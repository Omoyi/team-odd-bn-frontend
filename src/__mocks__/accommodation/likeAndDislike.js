const likeAndDislikeMockData = {
  likeInput: {
    status: 200,
    response: {
      message: 'like successfully',
      payload: [
        {
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        },
      ],
    },
  },

  likeExpectedResponse: [
    {
      type: 'LIKES_SUCCESS',
      payload: {
        message: 'like successfully',
        payload: [{
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        }],
      },
    },
  ],

  dislikeInput: {
    status: 200,
    response: {
      message: 'dislike successfully',
      payload: [
        {
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        },
      ],
    },
  },
  dislikeExpectedResponse: [
    {
      type: 'DISLIKES_SUCCESS',
      payload: {
        message: 'dislike successfully',
        payload: [{
          LikesDislikes: {
            dislikeCounter: 1,
            updatedAt: '2020-01-24T14:29:50.618Z',
            userId: 15,
          },
        }],
      },
    },
  ],

};
export default likeAndDislikeMockData;
