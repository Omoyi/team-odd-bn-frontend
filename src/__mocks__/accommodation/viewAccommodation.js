export const viewAccommodationProps = {

  allLikesDislakes: [
    {
      likeCounter: 1,
      dislikeCounter: 0,
    },
  ],
  allAccomodation: [
    {
      id: 45,
      userId: 15,
      name: 'Eric Palace',
      cityId: 2,
      address: 'Kigali',
      description: 'I am going there',
      googleCoordinates: '-22,-12',
      createdAt: '2020-01-24T14:29:50.618Z',
      updatedAt: '2020-01-24T14:29:50.618Z',
      accommodationRooms: [
        {
          id: 88,
          accommodationId: 45,
          name: 'Kackity',
          roomType: 'Double',
          createdAt: '2020-01-24T14:29:50.761Z',
          updatedAt: '2020-01-24T14:29:50.761Z',
        },
        {
          id: 89,
          accommodationId: 45,
          name: 'Serena',
          roomType: 'Single',
          createdAt: '2020-01-24T14:29:50.762Z',
          updatedAt: '2020-01-24T14:29:50.762Z',
        },
      ],
      imagesAccommodation: [
        {
          id: 62,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876184/Screen_Shot_2019-12-19_at_10.33.23_gnuiwa.png',
          createdAt: '2020-01-24T14:29:50.665Z',
          updatedAt: '2020-01-24T14:29:50.665Z',
        },
        {
          id: 61,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876134/34_bdjnqy.jpg',
          createdAt: '2020-01-24T14:29:50.657Z',
          updatedAt: '2020-01-24T14:29:50.657Z',
        },
      ],
      city: {
        id: 2,
        city: 'Kigali',
        createdAt: '2019-12-09T07:36:36.417Z',
        updatedAt: '2019-12-09T07:36:36.417Z',
      },
      ratings: [{
        rating: '2',
      }],
      LikesDislikes: { likeCounter: 0, dislikeCounter: 1 },
    },
  ],
  pageData: [
    {
      id: 45,
      userId: 15,
      name: 'Eric Palace',
      cityId: 2,
      address: 'Kigali',
      description: 'I am going there',
      googleCoordinates: '-22,-12',
      createdAt: '2020-01-24T14:29:50.618Z',
      updatedAt: '2020-01-24T14:29:50.618Z',
      accommodationRooms: [
        {
          id: 88,
          accommodationId: 45,
          name: 'Kackity',
          roomType: 'Double',
          createdAt: '2020-01-24T14:29:50.761Z',
          updatedAt: '2020-01-24T14:29:50.761Z',
        },
        {
          id: 89,
          accommodationId: 45,
          name: 'Serena',
          roomType: 'Single',
          createdAt: '2020-01-24T14:29:50.762Z',
          updatedAt: '2020-01-24T14:29:50.762Z',
        },
      ],
      imagesAccommodation: [
        {
          id: 62,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876184/Screen_Shot_2019-12-19_at_10.33.23_gnuiwa.png',
          createdAt: '2020-01-24T14:29:50.665Z',
          updatedAt: '2020-01-24T14:29:50.665Z',
        },
        {
          id: 61,
          accommodationId: 45,
          imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876134/34_bdjnqy.jpg',
          createdAt: '2020-01-24T14:29:50.657Z',
          updatedAt: '2020-01-24T14:29:50.657Z',
        },
      ],
      city: {
        id: 2,
        city: 'Kigali',
        createdAt: '2019-12-09T07:36:36.417Z',
        updatedAt: '2019-12-09T07:36:36.417Z',
      },
      ratings: [{
        rating: '2',
      }],
      LikesDislikes: { likeCounter: 0, dislikeCounter: 1 },
    },
  ],
  viewActionAccommodation: jest.fn(),
  accommodation: {
    display: {
      deleteComment: 'none',
      editTrip: 'none',
      currentPopUp: 'editTrip',
      bookAccommodation: 'none',
    },
    allAccomodation: [
      {
        id: 45,
        userId: 15,
        name: 'Eric Palace',
        cityId: 2,
        address: 'Kigali',
        description: 'I am going there',
        googleCoordinates: '-22,-12',
        createdAt: '2020-01-24T14:29:50.618Z',
        updatedAt: '2020-01-24T14:29:50.618Z',
        accommodationRooms: [
          {
            id: 88,
            accommodationId: 45,
            name: 'Kackity',
            roomType: 'Double',
            createdAt: '2020-01-24T14:29:50.761Z',
            updatedAt: '2020-01-24T14:29:50.761Z',
          },
          {
            id: 89,
            accommodationId: 45,
            name: 'Serena',
            roomType: 'Single',
            createdAt: '2020-01-24T14:29:50.762Z',
            updatedAt: '2020-01-24T14:29:50.762Z',
          },
        ],
        imagesAccommodation: [
          {
            id: 62,
            accommodationId: 45,
            imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876184/Screen_Shot_2019-12-19_at_10.33.23_gnuiwa.png',
            createdAt: '2020-01-24T14:29:50.665Z',
            updatedAt: '2020-01-24T14:29:50.665Z',
          },
          {
            id: 61,
            accommodationId: 45,
            imageUrl: 'http://res.cloudinary.com/victorkarangwa4/image/upload/v1579876134/34_bdjnqy.jpg',
            createdAt: '2020-01-24T14:29:50.657Z',
            updatedAt: '2020-01-24T14:29:50.657Z',
          },
        ],
        city: {
          id: 2,
          city: 'Kigali',
          createdAt: '2019-12-09T07:36:36.417Z',
          updatedAt: '2019-12-09T07:36:36.417Z',
        },
        ratings: [{
          rating: '2',
        }],
        LikesDislikes: { likeCounter: 0, dislikeCounter: 1 },
      },
    ],
    allLikesDislakes: [
      {
        likeCounter: 1,
        dislikeCounter: 0,
      },
      {
        likeCounter: 1,
        dislikeCounter: 1,
      },
    ],
    pageNo: 0,
    itemsPerPage: 5,
    changePageNo: jest.fn(),
  },
  likeAction: jest.fn(),
  dislikeAction: jest.fn(),
  statusInputAction: jest.fn(),
};

export const accommodationReducer = {
  successAccomodationReducer: {
    accommodations: ['data'],
    accommodationErrors: [],
    allLikes: [],
  },
  errorAccomodationReducer: {
    accommodations: [],
    accommodationErrors: ['Error'],
    allLikes: [],
  },
  likesDislikesAccomodationReducer: {
    accommodations: [],
    accommodationErrors: [],
    allLikes: [
      {
        likeCounter: 1,
        dislikeCounter: 0,
      },
    ],
  },
};
