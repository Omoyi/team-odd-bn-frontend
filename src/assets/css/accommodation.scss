@import './App.scss';
@import './base.scss';

.center-container {
  display: flex;
  flex-direction: column;
  margin: 10px auto;
  width: 620px;
  height: 750px;
  .page-header {
    margin-bottom: 10px;
    margin-top: 10px;
  }
  .accommodation-container {
    width: 620px;
    height: 750px;
    border: 1px solid $logo-color;
    .inputbox {
      display: flex;
      flex-direction: column;
      .name {
        margin: 10px auto;
      }
      .city {
        margin: 10px auto;
      }
      .googleCoordinates {
        margin: 10px auto;
      }
      .address {
        margin: 10px auto;
      }
      .description {
        margin: 30px auto 10px auto;
      }
      .aligned-container {
        margin: 10px auto;
        width: 435px;
      }
      .upload-btn {
        margin-top: 10px;
        .uploaded-photos {
          margin-top: 14px;
        }
      }
      .btn-long {
        width: 140px;
        margin-left: 1px !important;
        @extend %btn-long;
        text-align: center;
      }
      button {
        width: 160px;
        height: 40px;
        font-size: 16px;
        background-color: #c4c4c4;
      }
      input {
        width: 435px;
        height: 45px;
        margin-top: 10px;
        font-size: 15px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
        padding: 0px 0px 0px 10px!important;
        margin-right: 0px !important;
        font-family: muli;
      }
      select {
        width: 435px;
        height: 30px;
        margin-top: 10px;
        font-size: 15px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
        margin-right: 0px !important;
                padding: 0px 0px 0px 10px!important;
      }
      textarea {
        width: 435px;
        height: 60px;
        margin-top: 10px;
          font-size: 14px;
        border: 1px solid #5c5c5c;
        border-radius: 2px;
                padding: 0px 0px 0px 10px!important;
      }
    }
  }
}

@media screen and (min-width: 1024px) and (max-width: 1223px) {
}
@media screen and (min-width: 768px) and (max-width: 1000px) {
}

@media screen and (min-width: 324px) and (max-width: 1024px) {
}
@media screen and (max-width: 768px) {
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin-top: 40px !important;
          textarea {
            width: 230px;
                      
          }
        }
        .aligned-container {
          margin: 10px auto;
          width: 385px;
        }
        .upload-btn {
          margin-left: 6px;
        }
        .btn-long {
          margin-left: 5px !important;
        }
        button {
          width: 140px;
          height: 30px;
          background-color: #c4c4c4;
        }
        input {
          width: 100%;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          margin-bottom: 5px;
          font-family: muli;
        }
        select {
          width: 235px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
                    margin-bottom: 10px;
        }

      }
    }
  }
}
@media (max-width: 768px) and (min-width: 360px) {
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin: auto;
          textarea {
            width: 380px;
          }
        }
        input {
          width: 380px;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          padding-left: 100px;
          font-family: muli;
        }
        select {
          width: 380px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
        }
      }
    }
  }
}
@media screen and (max-width: 380px){
  .center-container {
    display: flex;
    flex-direction: column;
    margin: 10px auto;
    width: 100%;
    height: 700px;
    .page-header {
      margin-bottom: 10px;
      margin-left: 5px;
    }
    .accommodation-container {
      width: 100%;
      height: 700px;
      border: 1px solid $logo-color;
      .inputbox {
                margin-top: 30px;
        display: flex;
        flex-direction: column;
        margin-top: 20px;
        .name {
          margin: auto;
        }
        .city {
          margin: auto;
        }
        .googleCoordinates {
          margin: auto;
        }
        .address {
          margin: auto;
        }
        .description {
          margin: auto;
          textarea {
            width: 230px;
          }
        }
        .aligned-container {
          margin: 10px auto;
          width: 385px;
        }
        .upload-btn {
          margin-left: 6px;
        }
        .btn-long {
          margin-left: 5px !important;
        }
        button {
          width: 140px;
          height: 30px;
          background-color: #c4c4c4;
        }
        input {
          width: 100%;
          height: 45px;
          margin-top: 10px;
          font-size: 15px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
          font-family: muli;
        }
        select {
          width: 235px;
          height: 30px;
          margin-top: 10px;
          font-size: 10px;
          border: 1px solid #5c5c5c;
          border-radius: 2px;
        }
      }
    }
  }
}