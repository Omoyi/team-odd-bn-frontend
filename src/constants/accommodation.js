
export const createAccommodationList = [
  {
    input: 'Name',
    className: 'name',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Google coordinates',
    className: 'googleCoordinates',
    type: 'text',
    title: 'Name should be valid',
  },
  {
    input: 'Address',
    className: 'address',
    type: 'text',
    title: 'address should be valid',
  },
];
export const modalInput = [
  {
    input: 'Room name',
    className: 'room-name',
    type: 'text',
    title: 'Room name should be valid',
  },
];
