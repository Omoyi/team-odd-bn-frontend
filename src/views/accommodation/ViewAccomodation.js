import React, { Component } from 'react'
import {connect} from 'react-redux';
import StarRatings from 'react-star-ratings'
import Dashboard from '../Dashboard/sidebar/index'
import '../../assets/css/accommodation/allAccommodation.scss'
import roomBedIcon from '../../assets/images/room_bed/room-bed_32.png'
import roomMapIcon from '../../assets/images/room_map/map_32.png'
import likeIcon from '../../assets/images/like_icon/thumbs-up_32.png'
import disLikeIcon from '../../assets/images/dislike_icon/thumbs-down_32.png'
import findAverageRates from '../../helpers/averageRatings'
import { Paginate, PaginationStyle } from '../../helpers/Paginate'
import newAccommodation from '../../helpers/joinLikesAccomodation'
import PopUp from '../../components/PopUp'
import { fetchRequestsAction } from '../../redux/actions/tripsActions/fetchRequests'
import { likeAction, dislikeAction } from '../../redux/actions/likeAndDislikeAction';
import { viewActionAccommodation } from '../../redux/actions/allAccommodation';
import BookAccommodation from '../../views/accommodation/BookAccommodation'
import { Link } from 'react-router-dom'
import '../../assets/css/accommodation/allAccommodation.scss';
import { checkPrevilege } from '../../helpers/checkPrevilege';

export class viewAllAccomodation extends Component {
  async componentDidMount() {
    const { fetchRequestsAction } = this.props
    const {allAccomodation} = this.props.accommodation
   this.setState({allAccomodation})
    await fetchRequestsAction()
  }
  state = {
    currentIndex: 0,
     a: 1 ,
     likeInput: false,
     dislikeInput: false
  }

  render() {
    const {
      allAccomodation,
      allLikesDislakes,
      pageNo,
      itemsPerPage,
      changePageNo,
      display,
      popUpAction,
      updateBookAccommodationInput,
    } = this.props.accommodation    
    const totalAccommodation = newAccommodation(
      allAccomodation,
      allLikesDislakes
    )
    const chunkPages = Paginate(totalAccommodation, itemsPerPage)
    const pageData = { ...chunkPages }[pageNo]
    const averageRates = findAverageRates(pageData)

    this.refs.mytrips &&
      this.refs.mytrips.children.length > 0 &&
      PaginationStyle(this.refs.mytrips.children, pageNo)

    return (
      <Dashboard>
        <PopUp
          popUp={<BookAccommodation id={23} commentId={3} />}
          display={display.bookAccommodation}
        />
        <div className="accommodation-container">
          {
            checkPrevilege(4) && (
              <div className="add_new_accommodation">
                <Link to="/accommodation/create">New accommodation</Link>
              </div>
            )
          }
          <div className="accomodation-title">
            Barefoot accomodations
          </div>

          {pageData && pageData.length !== 0 && pageData.map((items, index) => (
            <div key={index} className="accommodation-data">
              <div className="accommodation-right-side">
              <Link className="accommodation-item-container" to={`accommodation/${items.id}`}>
                <img src={items.imagesAccommodation[0].imageUrl} alt="accomodation" />
              </Link>
              </div>
              <div className="accommodation-left-side">
                <h2 className="accomodation-content-title">{items.name}</h2>
                <div className="accomodation-content">
                  <div className="accomodation-content-side-a">
                    <section className="common-content">
                      <img src={roomBedIcon} alt="rooom icon" />
                      <span>
                        {items.accommodationRooms.length}
                        {' '}
available rooms
                      </span>
                    </section>
                    <section className="common-content">
                      <img src={roomMapIcon} alt="map icon" />
                      <span>{items.address}</span>
                    </section>
                    <section className="common-content like-dislike">
                      <div className="like-dislike-content">
                        <img src={likeIcon} alt="like icon" />
                      </div>
                      <div className="like-dislike-content">
                        <img className="dislike_icon" src={disLikeIcon} alt="dislike icon" />
                      </div>
                    </section>
                    <section className="common-content likes_dislikes_data">
                      <span className="likes_side">{items.LikesDislikes.likeCounter}</span>
                      <span className="dislikes_side">{items.LikesDislikes.dislikeCounter}</span>
                    </section>
                  </div>
                  <div className="accomodation-content-side-b">
                    <section className="rate-content">
                      <p className="average-rate">{averageRates[index] === 0 ? 0 : averageRates[index].toFixed(1)}</p>
                      <main>
                        <StarRatings
                          rating={averageRates[index]}
                          starRatedColor="#33c6f3"
                          starDimension="20px"
                          starSpacing="1px"
                        />
                      </main>
                    </section>
                    <p className="all-reviewes">
                      {items.ratings.length}
                      {' '}
reviews
                    </p>
                    <button
                        id="showBookingPopUp"
                        onClick={async () => {
                          await updateBookAccommodationInput({
                            accommodationId: items.id,
                            city:items.city.city
                          })
                          popUpAction({
                            currentPopUp: 'bookAccommodation',
                            bookAccommodation: 'flex',
                          })
                        }}
                        className="check-in"
                        type="button"
                      >
                        Check-in
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            ))}
          <div className="corasselButtons">
            <div className="pageArrows">
              <button
                type="button"
                onClick={() => changePageNo(pageNo - 1 < 0 ? 0 : pageNo - 1)}
              >
                <a href={`#${pageNo - 1}`}> &#60;&#60;</a>
              </button>
            </div>
            <div className="pageButtons" ref="mytrips">
              {chunkPages.map((item, index) => (
                <div key={index} id={`${index}`}>
                  <button
                    type="button"
                    onClick={() => {
                      const c = this.state.currentIndex > index ? -1 : 1
                      this.setState({
                        currentIndex: index,
                        a: c,
                      })
                      changePageNo(index)
                    }}
                  >
                    <a href={`#${index + this.state.a}`}>{index + 1}</a>
                  </button>
                </div>
              ))}
            </div>
            <div className="pageArrows">
              <button
                type="button"
                onClick={() =>
                  changePageNo(
                    pageNo === chunkPages.length - 1 ? pageNo : pageNo + 1
                  )
                }
              >
                <a href={`#${pageNo}`}> &#62;&#62;</a>
              </button>
            </div>
          </div>
        </div>
      </Dashboard>
    )
  }
}

export const mapStateToProps = ({userLikesAndDislike}) => ({
  likeDislikeState:userLikesAndDislike,
});

export default connect(mapStateToProps, { likeAction,viewActionAccommodation, dislikeAction, fetchRequestsAction })(viewAllAccomodation);
