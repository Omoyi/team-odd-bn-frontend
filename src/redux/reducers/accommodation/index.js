import { combineReducers } from 'redux';
import createAccommodation from './createAccommodationReducer';

export default combineReducers({
  createAccommodation,
});
