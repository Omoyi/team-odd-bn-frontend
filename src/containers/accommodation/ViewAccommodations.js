import React, { Component } from 'react';
import { connect } from 'react-redux';
import ViewAccomodation from '../../views/accommodation/ViewAccomodation';
import { viewActionAccommodation,getAllLikesDislikes } from '../../redux/actions/allAccommodation';
import '../../components/Spinner';
import Spinner from '../../components/Spinner';
import {changePageNo} from  '../../redux/actions/PaginateAction'
export class ViewAccommodations extends Component {
  componentDidMount() {
    const { viewActionAccommodation } = this.props;
    viewActionAccommodation();
  }

  accommodation = () => {
    const { allAccomodation, allLikesDislakes,itemsPerPage, pageNo, changePageNo,likeDislikeState } = this.props;    
    return {
      allAccomodation: allAccomodation,
      allLikesDislakes,
      itemsPerPage,
      pageNo,
      changePageNo,
      likeDislikeState,
      
    }
  }

  render() {
    const { allLikesDislakes } = this.props;        
    return (
      <>
        { allLikesDislakes.length === 0 && <Spinner/> }
        { allLikesDislakes.length !== 0 && <ViewAccomodation accommodation={this.accommodation()} /> }
      </>
    );
  }
}

export const mapStateToProps = (state) => ({
  allAccomodation: state.allAccomodation.accommodations,
  errorAccomodation: state.allAccomodation.accommodationErrors,
  allLikesDislakes: state.allAccomodation.allLikes,
  pageNo:state.pagination.pageNo,
  itemsPerPage:state.pagination.itemsPerPage,
  likeDislikeState: state.userLikesAndDislike

});

export default connect(mapStateToProps, { viewActionAccommodation,changePageNo, getAllLikesDislikes })(ViewAccommodations);
