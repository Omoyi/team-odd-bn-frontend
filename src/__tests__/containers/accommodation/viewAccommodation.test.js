import React from 'react';
import { shallow } from 'enzyme';
import { ViewAccommodations } from '../../../containers/accommodation/ViewAccommodations';
import { viewAccommodationProps } from '../../../__mocks__/accommodation/viewAccommodation';
import { viewAllAccomodation as ViewAllAccomodationView } from '../../../views/accommodation/ViewAccomodation';

describe('Expect to render accommodation container', () => {
  const component = shallow(<ViewAccommodations {...viewAccommodationProps} />);
  it('Should render without problem', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Expect to load accommodation', () => {
  const component = shallow(<ViewAllAccomodationView {...viewAccommodationProps} />);
  it('Should load accommodation', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Expect to mock accommodation state', () => {
  const component = shallow(<ViewAllAccomodationView {...viewAccommodationProps} />);
  // it('Should simulate on change paginate by arrow function', () => {
  //   const addButton = component.find('button').at(0);
  //   const showBookingPopUp = component.find('#showBookingPopUp');
  //   showBookingPopUp.simulate('click');
  //   addButton.simulate('click');
  // });

  // it('Should simulate on change paginate by left arrow change state function', () => {
  //   const addButton = component.find('button').at(1);
  //   addButton.simulate('click');
  // });
  // it('Should simulate on change paginate by  rightarrow change state function', () => {
  //   const addButton = component.find('button').at(2);
  //   addButton.simulate('click');
  // });
  // it('Should simulate on change paginate by changing no', () => {
  //   const addButton = component.find('button').at(3);
  //   addButton.simulate('click');
  // });

  // it('Should simulate on click img dislike and remove like', () => {
  //   const removeImg = component.find('.dislike_icon');
  //   removeImg.simulate('click');
  // });
});
