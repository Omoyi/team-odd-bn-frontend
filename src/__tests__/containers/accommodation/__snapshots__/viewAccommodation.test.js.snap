// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to load accommodation Should load accommodation 1`] = `ShallowWrapper {}`;

exports[`Expect to render accommodation container Should render without problem 1`] = `ShallowWrapper {}`;
